src/test/groovy/lang/BindingScript.java

```diff
@@ -0,0 +1,32 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package groovy.lang;
+
+/**
+ * A Script which requires a Binding passed in the constructor and disallows calling the default constructor.
+ */
+public abstract class BindingScript extends Script {
+    protected BindingScript() {
```

Comment:
Unfortunately the exact same `super()` call is generated, because `classNode.getSuperClass().getDeclaredConstructor(SCRIPT_CONTEXT_CTOR)` still returns null in `ModuleNode.createStatementsClass()` even though that constructor exists.

If i make the default constructor `private` i get `IllegalAccessError`:

    java.lang.reflect.InvocationTargetException
        at TestBindingsInFieldInitializersWithConfigJavaBaseScript.main(TestBindingsInFieldInitializersWithConfigJavaBaseScript.groovy)
    Caused by: java.lang.IllegalAccessError: tried to access method groovy.lang.BindingScript.<init>()V from class TestBindingsInFieldInitializersWithConfigJavaBaseScript
	    at TestBindingsInFieldInitializersWithConfigJavaBaseScript.<init>(TestBindingsInFieldInitializersWithConfigJavaBaseScript.groovy)

If i remove the default constructor i get `NoSuchMethodError`:

    java.lang.reflect.InvocationTargetException
        at TestBindingsInFieldInitializersWithConfigJavaBaseScript.main(TestBindingsInFieldInitializersWithConfigJavaBaseScript.groovy)
    Caused by: java.lang.NoSuchMethodError: groovy.lang.BindingScript: method <init>()V not found
	    at TestBindingsInFieldInitializersWithConfigJavaBaseScript.<init>(TestBindingsInFieldInitializersWithConfigJavaBaseScript.groovy)